flowtorch/bijectors/autoregressive.py

```diff
@@ -55,15 +59,23 @@ def inverse(
             params.permutation
         )  # TODO: type-safe named buffer (e.g. "permutation") access
         # TODO: Make permutation, inverse work for other event shapes
+        log_detJ = 0.0
         for idx in cast(torch.LongTensor, permutation):
-            x_new[..., idx] = self._inverse(y, x_new.clone(), context)[..., idx]
+            _params = params(x_new, context=context)
+            out = self._inverse(y, params=_params)
+            x_new[..., idx] = out[0][..., idx]
+            _log_detJ = out[1]
+            log_detJ += _log_detJ
```

Comment:
I think this may be a bug... Should `log_detJ` accumulate, or should it be the single value of _log_detJ once `x` has been fully calculated?

Reply:
Also, I think we previously used `x_new.clone()` for a specific reason although I can't remember why... It may introduce a bug by removing `.clone()` - we'll check on the unit tests!

Reply:
> I think this may be a bug... Should `log_detJ` accumulate, or should it be the single value of _log_detJ once `x` has been fully calculated?

Not sure but if you want to compute 
`log_detJ = log(abs(det(F(x))))` with `F(x) = [f(x[0]), f(x[1]), ..., f(x[-1])]`
you have `log_detJ = log(abs(det(J(f(x[0]))))) + log(abs(det(J(f(x[1]))))) + ... + log(abs(det(J(f(x[-1])))))`
no?

Reply:
> Also, I think we previously used `x_new.clone()` for a specific reason although I can't remember why... It may introduce a bug by removing `.clone()` - we'll check on the unit tests!

yes the reason is that the tensor is changed in place and messes up with backprop. I modified that.